commit e03dc9efdb9577455213d0faedcce18597c6174c (refs/changes/78/478/2)
Author: Alex Wilson <alex.wilson@joyent.com>
Date:   2016-09-16T16:34:56+00:00 (3 years, 1 month ago)
    
    joyent/node-cueball#22 ConnectionFSM states 'error' and 'idle' do not handle 'closeAsserted'
    Reviewed by: Cody Mello <cody.mello@joyent.com>

diff --git a/lib/pool.js b/lib/pool.js
index 905bf5a..9043e32 100644
--- a/lib/pool.js
+++ b/lib/pool.js
@@ -258,6 +258,12 @@ ConnectionFSM.prototype.state_closed = function (on) {
 
 ConnectionFSM.prototype.state_error = function (on, once, timeout) {
 	this.validTransitions(['delay', 'closed']);
+
+	var self = this;
+	on(this, 'closeAsserted', function () {
+		self.gotoState('closed');
+	});
+
 	if (this.cf_conn && this.cf_conn.destroy)
 		this.cf_conn.destroy();
 	this.cf_conn = undefined;
@@ -367,6 +373,7 @@ ConnectionFSM.prototype.state_idle = function (on, once, timeout) {
 		this.cf_closeAfter = false;
 		this.cf_lastError = undefined;
 		this.gotoState('closed');
+		on(this, 'closeAsserted', function () { });
 		return;
 	}
 
diff --git a/package.json b/package.json
index f602dff..92efc2c 100644
--- a/package.json
+++ b/package.json
@@ -1,6 +1,6 @@
 {
   "name": "cueball",
-  "version": "0.4.0",
+  "version": "0.4.1",
   "description": "",
   "main": "lib/index.js",
   "dependencies": {
@@ -9,7 +9,7 @@
     "cmdutil": ">=1.0.0 <2.0.0",
     "extsprintf": ">=1.3.0 <2.0.0",
     "ipaddr.js": ">=1.1.0 <2.0.0",
-    "mooremachine": ">=1.4.0 <2.0.0",
+    "mooremachine": ">=1.4.2 <2.0.0",
     "named-client": "git+https://github.com/arekinath/node-named-client.git#v0.3.5",
     "node-uuid": ">=1.4.7 <2.0.0",
     "posix-getopt": ">=1.2.0 <2.0.0",
